### PR TITLE
[AGW] Remove -Werror compiler flag everywhere, add -Wextra

### PR DESCRIPTION
**Pull request by electronjoe**
# Summary

Per  [This](https://github.com/magma/magma/discussions/5273) GH Discussion, here we drop the Werror compiler flag.  In its place, we offer #5404 which is capable of Github Pull Request GCC Warning annotations for code reviewers.

With `-Werror` gone, we also turn up the warnings by adding `-Wextra` and remove some suppressed warnings.

There are some places where `-Wextra` is noisy - e.g. functions that have unused parameters because they are callback driven and have to contain a specified shape.  We should be able to mitigate these warnings by appropriate modification of the unused parameter declaration. E.g. in C++ as advocated by [Herb Sutter](https://herbsutter.com/2009/10/18/mailbag-shutting-up-compiler-warnings/) and in C by omitting the name or void casting.

```
void foo(int /*bar_unused*/) {
    ...
}
```

or

```
void foo(int bar_unused) {
    // bar_unused is not needed for this function, but must exist to meet interface expectations
    (void)bar_unused;
    ...
}
```

Signed-off-by: Scott Harrison Moeller <smoeller@fb.com>

### CODE REVIEW

File: lte/gateway/c/session_manager/CMakeLists.txt
Comment (uri200) [inline lte/gateway/c/session_manager/CMakeLists.txt]: Hey @electronjoe, just curious, but per your comment on one of the linked PRs, it was mentioned you observed issues with mconfig while building `test_session_manager` on your docker container. 

Are you observing the same on our agw Vagrant vm when you do:
```
vagrant@magma-dev:~/magma/lte/gateway$ make test_session_manager
```

If not, maybe the issue is different gcc versions?

**Comment by rdefosse** [comment]
MAGMA-OAI-TESTS

File: lte/gateway/c/session_manager/CMakeLists.txt
Comment (electronjoe) [reply lte/gateway/c/session_manager/CMakeLists.txt]: Hi @uri200! Thanks for commenting.

> Hey @electronjoe, just curious, but per your comment on one of the linked PRs, it was mentioned you observed issues with mconfig while building `test_session_manager` on your docker container.
> 
> Are you observing the same on our agw Vagrant vm when you do:
> 
> ```
> vagrant@magma-dev:~/magma/lte/gateway$ make test_session_manager
> ```

Yes, In attempting to migrate to Ubuntu 20.04 I'm currently blocked by -Wdeprecated warnings on **protoc compiled protobuf outputs** that are converted to `-Werror`.  This does not occur in the Ubuntu 18.04 based vagrant VM - but note that we are migrating actively this quarter to Ubuntu 20.04. There is a Ubuntu 20.04 Vagrant floating around, I have not tracked it down yet to determine if and how this was addressed.

> If not, maybe the issue is different gcc versions?

This is gcc 9.3 (the baseline for Ubuntu 20.04) - but the issue is with the -Werror flag being applied to protobuf compiles (which is not the case in our older Ubunto 18.04 Vagrant VMs).  Ultimately this points to a problem with our CMake configurations  [See technical debt enumerated in #5222] and / or a problem with Cmake (less likely) - triggered by moving forward with the CMake bundled with Ubuntu 20.04.  Looking at how we compile these protos - it's not surprising to me that this is a tangle (compiler flags leaking / inconsistently applied across build targets and build environments).

Another option is to remove -Wdeprecated (which is what blocks the above - but would need additional fixups to allow Clang to build without `-Werror`).  Having `-Wdeprecated` and `-Werror` is super inconsistent.  What is the point of annotating deprecation if you block compile time until it's removed entirely from the codebase.

edit: I'm realizing - and you were probably indicating above - that GCC 9.3 may have added Deprecation warnings to `-Wall`, and that could be the origin of this particular issue.  We could add `-Wno-deprecated-declarations` to ignore deprecation warnings.  Though this seems to negate the value of having deprecation annotations.  And this doesn't address the Clang build issue (with 100's of new warnings that block builds).  Ultimately I just don't feel `-Werror` is healthy - and per some of my references in GH Discussion it seems that is not a unique view.

**Review by themarwhal — APPROVED** [review]

**Review by ssanadhya — APPROVED** [review]

File: lte/gateway/c/session_manager/CMakeLists.txt
Comment (uri200) [reply lte/gateway/c/session_manager/CMakeLists.txt]: @electronjoe thanks for the detailed description.
I agree with the -werror [article](https://embeddedartistry.com/blog/2017/05/22/werror-is-not-your-friend/#:~:text=In%20case%20you%20have%20been,behind%20enabling%20%2DWerror%20are%20benevolent.&text=%2DWerror%20creates%20a%20project%20dependency%20on%20a%20specific%20compiler%20version.) you mentioned.
My concerned was just for session_manager: if it worked so far and it was not giving us errors, it would be nice to make it work in Ubuntu focal. But maybe it doesn't make sense anymore

Will take a look at Focal VM and see if, at least for session_manager, we can keep it.

BTW, proto generation is kind of messy. Sometimes it is even preventing the IDE to use the autocomplete properly with the protos, which makes the developer experience very poor. So any change on that would be awesome!

Comment (electronjoe) [reply lte/gateway/c/session_manager/CMakeLists.txt]: @uri200 you prompted me to get GitHub GCC Warning annotations working for Pull Request code reviews.  I think this clears the way quite well for this PR to land, removing `-Werror`?

See PR #5404

**Comment by electronjoe** [comment]
MAGMA-OAI-TESTS

**Review by koolzz — APPROVED** [review]

File: lte/gateway/c/session_manager/CMakeLists.txt
Comment (uri200) [reply lte/gateway/c/session_manager/CMakeLists.txt]: Sound like a good solution to unblock the build and still keep the warnings visibility👍

**Review by rdefosse — COMMENTED** [review]
OK for me

**Pull request merged**
